### PR TITLE
Fix #11716: Failure in Dynamic Image ServeView

### LINKED ISSUE #11716: Dynamic Image Serve View is failing
### Issue Summary

When using the Dynamic image serve view, from the documentation (https://docs.wagtail.org/en/stable/advanced_topics/images/image_serve_view.html) the URL's are generated, but they cannot be accessed from a GET request. 

### Steps to Reproduce

1. Follow setup instructions here https://docs.wagtail.org/en/stable/advanced_topics/images/image_serve_view.html
2. Add an `image = models.ForeignKey( get_image_model(), ...)` field to the HomePage class. Make migrations and apply them.
3. Use the dynamic serve view in the template. Example code below:

```django+html
{% extends "base.html" %}
{% load wagtailimages_tags %}

{% block content %}
    {% image_url page.image "fill-150x150" as this_works %}
    {{ this_works }}

    {# This does not work #}
    <img src="{{ this_works }}" alt="">
{% endblock content %}
```

### Other details
Terminal output is:
```bash
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/wsgiref/handlers.py", line 138, in run
    self.finish_response()
  File "/Users/kalobtaulien/Sites/tmp/svgtest/.venv/lib/python3.11/site-packages/django/core/servers/basehttp.py", line 173, in finish_response
    super().finish_response()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/wsgiref/handlers.py", line 183, in finish_response
    for data in self.result:
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/wsgiref/util.py", line 24, in __next__
    data = self.filelike.read(self.blksize)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: read of closed file
```

What the browser outputs:
image

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: **yes** 

### Technical details

Python: 3.11
Wagtail: 6.0.1
Django: 5.0.2

### Working on this

Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

### PR DESCRIPTION
**Pull request by SebCorbin**
This is a rework of https://github.com/wagtail/wagtail/pull/11825 (as I don't have access to the original branch)

### CODE REVIEW

File: wagtail/images/tests/tests.py [hunk 1]
```diff
@@ -359,6 +359,9 @@ def test_get(self):
         self.assertEqual(response.status_code, 200)
         self.assertTrue(response.streaming)
         self.assertEqual(response["Content-Type"], "image/png")
+        for _byte in response:
+            # Ensure the file can actually be read
+            self.assertTrue(_byte)
```
Comment (RealOrangeOne) [inline wagtail/images/tests/tests.py:364]: Nit: Could we do a better assertion than just the truthyness of the bytes? Perhaps passing it into `willow` and checking it looks like a sensible image?

I realise that's slightly out of scope of this PR, but it'd be good to confirm what we get back is actually an image and not garbage data.

Similarly, it'd then be great to add similar assertions to the related test cases.

File: wagtail/images/tests/tests.py [hunk 2]
```diff
@@ -412,6 +437,12 @@ def test_get_with_extra_component(self):
         self.assertEqual(response.status_code, 200)
         self.assertTrue(response.streaming)
         self.assertEqual(response["Content-Type"], "image/png")
+        # Ensure the file can actually be read
+        with BytesIO() as buffer:
+            for _bytes in response:
+                buffer.write(_bytes)
+            image = willow.Image.open(buffer)
+        self.assertIsInstance(image, PNGImageFile)
```
Comment (RealOrangeOne) [inline wagtail/images/tests/tests.py:445]: Nit: I don't think you need the loop here, although I may be wrong.
```suggestion
        # Ensure the file can actually be read
        image = willow.Image.open(response.content)
        self.assertIsInstance(image, PNGImageFile)
```

Comment (SebCorbin) [reply wagtail/images/tests/tests.py:445]: Tried with `willow.Image.open(response.streaming_content)` but it's not quite a file, it's just a generator that can be consumed, so I got this error
```
Traceback (most recent call last):
  File "/home/sebcorbin/projects/wagtail/wagtail/images/tests/tests.py", line 478, in test_get_with_serve_action
    image = willow.Image.open(response.streaming_content)
  File "/home/sebcorbin/.virtualenvs/wagtail/lib/python3.10/site-packages/willow/image.py", line 84, in open
    image_format = filetype.guess_extension(f)
  File "/home/sebcorbin/.virtualenvs/wagtail/lib/python3.10/site-packages/filetype/filetype.py", line 63, in guess_extension
    kind = guess(obj)
  File "/home/sebcorbin/.virtualenvs/wagtail/lib/python3.10/site-packages/filetype/filetype.py", line 28, in guess
    return match(obj) if obj else None
  File "/home/sebcorbin/.virtualenvs/wagtail/lib/python3.10/site-packages/filetype/match.py", line 30, in match
    buf = get_bytes(obj)
  File "/home/sebcorbin/.virtualenvs/wagtail/lib/python3.10/site-packages/filetype/utils.py", line 84, in get_bytes
    raise TypeError('Unsupported type as file input: %s' % type(obj))
TypeError: Unsupported type as file input: <class 'map'>
```

**Review by RealOrangeOne — APPROVED** [review]

**Comment by RealOrangeOne** [comment]
Merged in 76fb103f5489757f647bca44adf4640eb300f99d, thanks!

**Pull request closed without merging**